gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ReplicationData.java

```diff
@@ -0,0 +1,31 @@
+package gobblin.data.management.copy.replication;
+
+import gobblin.data.management.copy.CopyableDataset;
+import gobblin.dataset.DatasetsFinder;
+
+
+/**
+ * Interface Replication Data is used to describe the data which need to be replicated.
+ * 
+ * A ReplicationData is responsible for:
+ * <ul>
+ *  <li>Telling whether the data is source or replicas
+ *  <li>Specifying the replication name
+ *  <li>Specifying the replication location {@link ReplicationLocation}
+ *  <li>Specifying the {@link DatasetsFinder} which can find all data sets
+ * </ul>
+ * 
+ * @author mitu
+ *
+ * @param <T>
+ */
+public interface ReplicationData<T extends CopyableDataset> {
+
+  public boolean isSource();
+
+  public String getReplicationName();
+
+  public ReplicationLocation getReplicationLocation();
```

Comment:
The ReplicationLocation interface does not seem to declare any useful methods. It is unusable without a bunch of if () else statements with typecasts which then invoke implementation-specific behavior. This is not a proper use of interfaces. Interfaces should not require a use of typecasts.

Further, there might be more than one "locations" associated with the copy. For example, when copying Hive tables, we may need to know both the filesystem paths where to write the data files and the Hive DB/Table names.

Finally, "hdfs" is a bad name for a type. We should support any Hadoop FileSystem implementation.